examples/diff.c

```diff
@@ -238,6 +275,8 @@ static void parse_opts(struct opts *o, int argc, char *argv[])
 			o->format = GIT_DIFF_FORMAT_RAW;
 			o->diffopts.id_abbrev = 40;
 		}
+		else if (!strcmp(a, "--no-index"))
+			o->no_index = 0;
```

Comment:
We should catch incompatibilities between options. Like for example "--cached" doesn't make any sense with "--no-index", right?

Reply:
Done